.github/workflows/on-master-push.yaml

```
@@ -0,0 +1,18 @@
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
```

```suggestion
      - main
```
Also, rename the file itself